src/Xamarin.Android.Build.Tasks/Utilities/AssemblyBlob.cs

```diff
@@ -0,0 +1,391 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks
+{
+	abstract class AssemblyBlob
+	{
+		// The two constants below must match their counterparts in src/monodroid/jni/xamarin-app.hh
+		const uint BlobMagic = 0x41424158; // 'XABA', little-endian, must match the BUNDLED_ASSEMBLIES_BLOB_MAGIC native constant
+		const uint BlobVersion = 1; // Must match the BUNDLED_ASSEMBLIES_BLOB_VERSION native constant
+
+		// MUST be equal to the size of the BlobBundledAssembly struct in src/monodroid/jni/xamarin-app.hh
+		const uint BlobBundledAssemblyNativeStructSize = 6 * sizeof (uint);
```

Comment:
(Insert screaming for [`create-native-map`](https://github.com/mono/mono-tools/blob/main/create-native-map/man/create-native-map.1) here…. Pity that's not packaged anymore.)

Reply:
I suppose a more "in-tree" solution to ensuring that things are consistent would be to have `Xamarin.Android.Build.Tasks.csproj` have a `@(ProjectReference)` on `src/monodroid/monodroid.csproj`, and have `monodroid.csproj` write out a file containing the sizes of these structures?  Or vice-versa, have `monodroid.csproj` reference `Xamarin.Android.Build.Tasks.csproj`, and have XABT emit a `.h` file as part of its build?